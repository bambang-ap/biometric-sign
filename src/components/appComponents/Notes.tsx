import React from 'react';
import {Text, TouchableOpacity, View} from 'react-native';

import moment from 'moment';
import {useSetRecoilState} from 'recoil';

import {NotesProps} from '@appTypes/propsType.type';
import {Icon} from '@components';
import {FORMAT_DATE_VIEW} from '@constants';
import {atomNotes} from '@recoils';
import {prompt} from '@utils';

export default function Notes({id, title, notes, date, onPress}: NotesProps) {
  const setNotes = useSetRecoilState(atomNotes);

  function deleteNote() {
    prompt('Delete this note?', {
      onConfirm() {
        setNotes(prev => {
          const index = prev.findIndex(note => note.id === id);

          if (index >= 0) return prev.remove(index);

          return prev;
        });
      },
    });
  }

  return (
    <TouchableOpacity
      onPress={onPress}
      className="p-2 border border-black rounded-lg mb-2">
      <View className="flex-1 mb-4">
        <Text
          numberOfLines={1}
          ellipsizeMode="clip"
          className="text-black font-bold text-lg">
          {title}
        </Text>
        <Text className="text-black text-justify">{notes}</Text>
      </View>
      <View className="flex-row items-center">
        <Icon
          className="text-black text-sm"
          name="trash"
          onPress={deleteNote}
        />
        <Text
          className="text-black flex-1 text-right font-semibold"
          numberOfLines={1}>
          {moment(date).format(FORMAT_DATE_VIEW)}
        </Text>
      </View>
    </TouchableOpacity>
  );
}
