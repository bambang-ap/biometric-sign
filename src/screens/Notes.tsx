import React from 'react';
import {Text, TouchableOpacity} from 'react-native';

import {FlashList} from '@shopify/flash-list';
import {useRecoilValue} from 'recoil';

import AppScreen from '@appComp/AppScreen';
import Notes from '@appComp/Notes';
import {TNotes} from '@appTypes/app.zod';
import {RootStackList} from '@appTypes/navigators.type';
import {Icon} from '@components';
import {atomNotes} from '@recoils';
import {useStackNavigation} from '@utils/navigators';

export default function NotesScreen() {
  const data = useRecoilValue(atomNotes);
  const {navigation} = useStackNavigation();

  function navigate(notes: TNotes) {
    navigation.navigate(RootStackList.NoteDetail, notes);
  }

  function addNew() {
    navigation.navigate(RootStackList.NoteDetail);
  }

  return (
    <AppScreen>
      <FlashList
        data={data}
        ListEmptyComponent={
          <Text className="text-black">
            Noting to see here, please add some notes
          </Text>
        }
        renderItem={({item}) => {
          return <Notes {...item} onPress={() => navigate(item)} />;
        }}
      />
      <TouchableOpacity
        className="absolute bottom-5 right-5 w-14 h-14 bg-green-600 rounded-full items-center justify-center"
        onPress={addNew}>
        <Icon name="plus" className="text-white text-xl" />
      </TouchableOpacity>
    </AppScreen>
  );
}
