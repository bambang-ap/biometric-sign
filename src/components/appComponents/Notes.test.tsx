import React from 'react';

import {render} from '@testing-library/react-native';
import FullMockComponent from '__mocks__/@components';
import renderer from 'react-test-renderer';

import Notes from './Notes';

describe('Notes Component', () => {
  const TestComponent = () => {
    return (
      <FullMockComponent>
        <Notes
          id="id-1"
          notes="Lorem"
          title="title"
          date="2023-10-20 22:30:25"
        />
      </FullMockComponent>
    );
  };

  test('Notes render correctly', () => {
    render(<TestComponent />);
  });

  test('Notes snapshot', () => {
    const wrapper = renderer.create(<TestComponent />).toJSON();
    expect(wrapper).toMatchSnapshot();
  });
});
