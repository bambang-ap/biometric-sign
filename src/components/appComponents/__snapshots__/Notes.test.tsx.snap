// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Notes Component Notes snapshot 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "borderBottomColor": "#000",
      "borderBottomLeftRadius": 8,
      "borderBottomRightRadius": 8,
      "borderBottomWidth": 1,
      "borderLeftColor": "#000",
      "borderLeftWidth": 1,
      "borderRightColor": "#000",
      "borderRightWidth": 1,
      "borderTopColor": "#000",
      "borderTopLeftRadius": 8,
      "borderTopRightRadius": 8,
      "borderTopWidth": 1,
      "marginBottom": 8,
      "opacity": 1,
      "paddingBottom": 8,
      "paddingLeft": 8,
      "paddingRight": 8,
      "paddingTop": 8,
    }
  }
>
  <View
    style={
      [
        {
          "flexBasis": "0%",
          "flexGrow": 1,
          "flexShrink": 1,
        },
        {
          "marginBottom": 16,
        },
      ]
    }
  >
    <Text
      ellipsizeMode="clip"
      numberOfLines={1}
      style={
        [
          {
            "fontWeight": "700",
          },
          {
            "fontSize": 18,
            "lineHeight": 28,
          },
        ]
      }
    >
      title
    </Text>
    <Text
      style={
        [
          {
            "textAlign": "justify",
          },
        ]
      }
    >
      Lorem
    </Text>
  </View>
  <View
    style={
      [
        {
          "flexDirection": "row",
        },
        {
          "alignItems": "center",
        },
      ]
    }
  >
    <Text
      allowFontScaling={false}
      onPress={[Function]}
      selectable={false}
      style={
        [
          {
            "color": undefined,
            "fontSize": 12,
          },
          [
            {
              "fontSize": 14,
              "lineHeight": 20,
            },
          ],
          {
            "fontFamily": "FontAwesome5Free-Solid",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
          {
            "fontWeight": "900",
          },
        ]
      }
    >
      ï‡¸
    </Text>
    <Text
      numberOfLines={1}
      style={
        [
          {
            "flexBasis": "0%",
            "flexGrow": 1,
            "flexShrink": 1,
          },
          {
            "textAlign": "right",
          },
          {
            "fontWeight": "600",
          },
        ]
      }
    >
      20-10-2023 22:30:25
    </Text>
  </View>
</View>
`;
